src/theory/sets/theory_sets_rewriter.cpp

```diff
@@ -482,7 +482,16 @@ RewriteResponse TheorySetsRewriter::preRewrite(TNode node) {
   Kind k = node.getKind();
   if (k == kind::EQUAL)
   {
-    if(node[0] == node[1]) {
+    if (node[0] == node[1])
+    {
+      return RewriteResponse(REWRITE_DONE, nm->mkConst(true));
+    }
+    // check equality between empty sets (possibly of different types)
+    // e.g.  (= (as emptyset (Set Int)) (as emptyset (Set Real)))
+    if (node[0].getKind() == EMPTYSET && node[1].getKind() == EMPTYSET
+        && (node[0].getType().isSubtypeOf(node[1].getType())
```

Comment:
I believe the 3rd condition is redundant: it should hold for all well-typed equalities.